### PR TITLE
[WAIT til v.019] remove all instances of realtime

### PR DESCRIPTION
**Pull request by teonbrooks**
this PR aims to remove the realtime module from the package. it will soon live at 
`mne-tools/mne-realtime`

### CODE REVIEW

**Comment by teonbrooks** [comment]
cf. #6339

**Comment by massich** [comment]
Circle failure is unrelated, something changed in the config to use matplotlib.

But what I don't understand is why I cannot reproduce the `test_docstring_parameter` in travis and azure

**Comment by teonbrooks** [comment]
> But what I don't understand is why I cannot reproduce the `test_docstring_parameter` in travis and azure

found the culprit, it was this [line](https://github.com/mne-tools/mne-python/pull/6340/files#diff-b6feb560f189016f0cb9b9dcde27e9a4L38).

**Comment by massich** [comment]
Wait wait a sec. I just realized that we cannot kill all `mne.realtime` just like this! it has been there since `v0.7`. What we should do is to deprecate it. And I think it's the same to deprecate part of the module or the entire module. What would be great is to deprecate it here, so we earn 6 months. But I don't think that doing it in a hurry is a good idea.

**Comment by jasmainak** [comment]
Yo, I wrote on gitter but apparently my message didn't go through. 

What is the plan here? @teonbrooks are you willing to maintain the real-time module? How will testing work? The RtEpochs depends on BaseEpochs, so changes in BaseEpochs could potentially break RtEpochs. There have been many instances where this could be caught because it was part of the same repository but it's not clear to me how this would work in the new system. I'm not yet fully convinced that there is a benefit to removing code that doesn't change very rapidly but I'm willing to change my mind. :-)

We don't have many users but as far as I know, perhaps the group of @LauriParkkonen would be interested to know if the module is being moved out into its own repository.

**Comment by agramfort** [comment]
to merge this after the 0.18 release, the 0.18 release must deprecate the module.
I am fine with moving this to mne-realtime as it can simplify testing for mne-python
and it will allow to move faster with a smaller project.

**Comment by teonbrooks** [comment]
@jasmainak, I am planning to maintain it. I will like to try to build some more momentum with the BCI community for it. it's been often requested but has not been straightforward to use.
I have already begun the setup for testing and it all exists [here](https://github.com/teonbrooks/mne-realtime) at the moment. I will essentially pull out the testing that we got setup during the sprint and bring it there.

**Comment by jasmainak** [comment]
can we make sure before merging this stuff that what exists does not break in the new repository? That is, do the steps of 1) make an mne-tools/mne-realtime 2) setup the CIs and website before merging this? Does this need to make in by 0.18? If not, I would do those steps first so we don't have an unforeseen technical challenge before hurriedly deprecating this.

I agree a different repo can help move things faster! Also I'm all for a modular design so that things are simpler in mne. But just keep in mind the downsides -- I don't know who has the time for the real-time stuff. For the last 5 years, the only time things have move forward on it is when we were at a sprint. It took 2 years to get the pylsl stuff in. The main mne repo moves much faster, so if this repo doesn't move half as fast, it's going to start breaking things. But okay for me if @teonbrooks and @massich want to take charge

File: doc/python_reference.rst [hunk 1]
```diff
@@ -946,28 +946,6 @@ Functions that assist with decoding and model fitting:
    get_coef
 
 
-Realtime
-========
```
Comment (larsoner) [inline doc/python_reference.rst]: We should keep this section header and add a sentence linking to `mne-tools.github.io/mne-realtime` or wherever we put the docs for this new repo.

File: doc/whats_new.rst [hunk 2]
```diff
@@ -144,14 +144,10 @@ Changelog
 
 - Improved clicking in :meth:`mne.io.Raw.plot` (left click on trace toggles bad, left click on background sets green line, right click anywhere removes green line) by `Clemens Brunner`_
 
-- Add :class:`mne.realtime.LSLClient` for realtime data acquisition with LSL streams of data by `Teon Brooks`_ and `Mainak Jas`_
-
 - Add partial support for PyVista as a 3D backend using :func:`mne.viz.use_3d_backend` by `Guillaume Favelier`_
 
 - Add option ``ids = None`` in :func:`mne.event.shift_time_events` for considering all events by `Nikolas Chalas`_ and `Joan Massich`_
 
-- Add :class:`mne.realtime.MockLSLStream` to simulate an LSL stream for testing and examples by `Teon Brooks`_
```
Comment (larsoner) [inline doc/whats_new.rst]: Don't delete these, instead just make them double-backticks on the refs.

**Comment by larsoner** [comment]
Actually the way this PR is structured we shouldn't do this until 0.19 *is released*. So let's just close for now and reopen this after 0.19 as part of the standard deprecations.

**Pull request closed without merging**